pymeasure/experiment/results.py

```diff
@@ -376,20 +394,57 @@ def data(self):
                 pass  # All data is up to date
         return self._data
 
-    def reload(self):
-        """ Preforms a full reloading of the file data, neglecting
+    def reload(self, backupCount=None):
+        """ Performs a full reloading of the file data, neglecting
         any changes in the comments
+
+        :param backupCount: If a rotating file handler is used,
+                            how many backup files to load.
+                            If None, only load active file.
+                            If -1, load all. (Only loads consecutively numbered files.)
+                            (default: None)
         """
-        chunks = pd.read_csv(
-            self.data_filename,
-            comment=Results.COMMENT,
-            chunksize=Results.CHUNK_SIZE,
-            iterator=True
-        )
-        try:
-            self._data = pd.concat(chunks, ignore_index=True)
-        except Exception:
-            self._data = chunks.read()
+        # get data files
+        data_files = [ self.data_filename ]
+        if backupCount is None:
+            # needed to avoid comparison between None and integer
+            pass
+
+        elif backupCount > 0:
+            for i in range(backupCount):
+                filename = f'{self.data_filename}.{i+1}'
+                if os.path.isfile(filename):
+                    data_files.append(filename)
+
+        elif backupCount < 0:
+            i = 1
+            filename = f'{self.data_filename}.{i}'
+            while os.path.isfile(filename):
+                data_files.append(filename)
+                i += 1
+                file_name = f'{self.data_filename}.{i}'
```

Comment:
Now I come to think about it: I think this only works for the RotatingFileHandler case, not for the TimedRotatingFileHandler case. Maybe using `glob` with the base-filename would solve this.